### ISSUE TITLE
vslm ExtendDisk and InflateDisk only support the host-based tasks and not the vCenter tasks

### ISSUE THREAD

**Issue by protochron** [vmware/govmomi#2308]
It looks like when #2067 went in, the vslm `ExtendDisk` and `InflateDisk` methods only implemented the calls to `HostExtendDisk_Task` and `HostInflateDisk_Task` so they do not support calling  `ExtendDisk_Task` and `InflateDisk_Task` like many of the other methods in  `vslm.LocalObjectManager`.

It seems like both calls are missing something resembling:

```go
if m.isVC {
	res, err := methods.ExtendDisk_Task(ctx, m.c, &req)
	if err != nil {
		return nil, err
	}

	return object.NewTask(m.c, res.Returnval), nil
}
```

I've confirmed that `methods.ExtendDisk_Task` works if you build up the task manually with a `*vim25.Client`, so it seems like there shouldn't be a reason not to add support for both types of task unless there's some subtlety in the API that I'm missing.

**Comment by dougm** [comment]
Sounds right to me @protochron , do you want to submit a PR to add those?

cc @yashdesai93

**Comment by dougm** [comment]
Fixed in #2321 (thanks @protochron)
